### PR TITLE
Chat 280/validate remote chat list

### PR DESCRIPTION
**Pull request by crgbaumgart**
- Ensure that when a user explicitly leaves a chat (w/ enableSync: true) and reconnects, that the previously left channel is not present in session.chats

### CODE REVIEW

**Comment by crgbaumgart** [comment]
Codacy lies.

**Comment by ianjennings** [comment]
@crgbaumgart is lying. I tried to lint your stuff yesterday and got the same.

**Comment by crgbaumgart** [comment]
@ianjennings Please see line 687: 
```
syncChat = new ChatEngineClone.Chat(newChannel3);
```

**Comment by ianjennings** [comment]
I think that @codacy-bot doesn't like that you've only defined the var and never used it. 
You can do something stupid like ````syncChat.objectify()``` which returns JSON to resolve this. The proper solution is to create two lint configs, one for tests (where we can disable this) and one for src, where we can keep this enabled.

File: test/integration/main.test.js [hunk 1]
```diff
@@ -645,6 +645,53 @@ describe('remote chat list', () => {
...
+                ChatEngineSync.on('$.group.restored', () => {
+                    assert.isObject(ChatEngineSync.me.session.chats.system);
+                    assert.equal(ChatEngineSync.me.session.chats.custom, undefined);
+                    groupRestored = true;
+                });
+
+                // Await the population of $me.session.chats from enableSync
+                setTimeout(() => {
+                    if (groupRestored === true) {
+                        done();
```
[the remark this answers is not in the document]
Comment (crgbaumgart) [reply test/integration/main.test.js]: The way I've noticed sync working, theres about 3-4 requests that come in trying to establish the users session.chats state. This ensures that at least one state refresh message is valid and then further validates the remaining calls within that timeout. Its a little wonky, but I was unable to find a better alternative.

**Pull request merged**